base/set.jl

```diff
@@ -191,6 +191,63 @@ function unique(f::Callable, C)
 end
 
 """
+    unique!(A::AbstractVector)
+
+Remove duplicate items as determined by [`isequal`](@ref), then return the modified `A`.
+
+```jldoctest
+julia> unique!([1, 1, 1])
+1-element Array{Int64,1}:
+ 1
+
+julia> A = [7, 3, 2, 3, 7, 5];
+
+julia> unique!(A);
+
+julia> A
+4-element Array{Int64,1}:
+ 7
+ 3
+ 2
+ 5
+```
+"""
+function unique!(A::AbstractVector)
+    if isempty(A)
+        return A
+    elseif issorted(A) || issorted(A, rev=true)
+        # If A is sorted, then we only need to keep track of one element and add that to A
+        # every time that we see a new element.
+        count = 1
```

Comment:
To support arrays whose indices do not start at `1`, better use the same strategy as for the other branch via `m = start(idxs)`, etc. Then you can compute `count` at the end as `m-start(idxs)`.

Also, putting `@inbounds` before the `for` will enable it for the whole block, including `A[count]`.